### ISSUE TITLE
This package does not compile

### ISSUE THREAD

**Issue by patientplatypus** [spring-guides/gs-accessing-mongodb-data-rest#8]
I am getting the following error on running the complete package: 

2017-06-11 12:54:12.973  INFO 25961 --- [           main] org.mongodb.driver.cluster               : No server chosen by WritableServerSelector from cluster description ClusterDescription{type=UNKNOWN, connectionMode=SINGLE, serverDescriptions=[ServerDescription{address=localhost:27017, type=UNKNOWN, state=CONNECTING, exception={com.mongodb.MongoSocketOpenException: Exception opening socket}, caused by {java.net.ConnectException: Connection refused (Connection refused)}}]}. Waiting for 30000 ms before timing out

I suspect it is related to this stackoverflow post: https://stackoverflow.com/questions/31933821/cant-connect-to-local-monogodb-from-java. 

So essentially your code has a race condition between the mongodb database and the code (which is an error that is not always reproducible, as it depends on the speed and operating environment of the users computer). I have had the same issue with your other mongodb tutorial and so will be opening an issue there as well. It looks like to solve this it will require a refactor, but as I am new to java it is something that I cannot do on my own. Please fix!

**Comment by dsyer** [comment]
Works for me, and we have a CI build (actually 2 now) that is also green. What was the command that you ran, and where? Was mongodb running last calls on the normal port?

**Comment by patientplatypus** [comment]
Ok. 
So what I did was to git clone the repo, cd into the "complete directory", and then in terminal ran "mvn clean package". After that I got the error described above. If I run other examples that you have put up then I get compiled and working code, so I am pretty sure that my environment is set up correctly. If it helps at all here is a link to my terminal code: https://hastebin.com/ojunexawos.scala

The errors start on 326 to 333, and are explained in more detail in the rest of the output (the surefire error logs are the same). I do not see anything in the tutorial on doing any special configuring for the paths of the mongodb connections, so I assumed that they should run out of the box, but they do not. I am not sure what I am doing wrong, but it seems like it is not compiling as it should.
